circuit_knitting_toolbox/circuit_cutting/qpd/qpd.py

```diff
@@ -42,6 +43,63 @@
 from ...utils.iteration import unique_by_id
 
 
+class WeightType(Enum):
+    """Type of weight associated with a QPD sample."""
+
+    #: A weight given in proportion to its exact weight
+    EXACT = 1
+
+    #: A weight that was determined through some sampling procedure
+    SAMPLED = 2
+
+
+def generate_qpd_samples(
+    qpd_bases: Sequence[QPDBasis], num_samples: int = 1000
```

Comment:
I wonder if at some point we should remove this default `num_samples`.  It would be so easy to mistakenly write code like
```python
num_samples = 10_000
samples = generate_qpd_samples([gate.basis for gate in qpd_gates])
```

Reply:
Ya it's a good thought. Does the function make sense without this argument being made explicit? I could see it either way.

Reply:
I talked to Matthew Treinish about this, and he convinced me that having the default is nice.  It is often convenient to have a reasonable default number of `shots` throughout Qiskit, and `num_samples` is very similar to shots.